modules/user/user.service.js

```diff
@@ -91,12 +130,35 @@ class UserService {
     return checkedUser;
   }
 
-  async getAllUsers({ filter }) {
-    const filters = this.filterItems(filter);
+  async getAllUsers({ filter, pagination, sort }) {
+    let filteredItems = this.filterItems(filter);
+    let aggregatedItems = this.aggregateItems(filteredItems, pagination, sort);
 
-    const items = await User.find(filters);
+    const users = await User.aggregate([
+      {
+        $addFields: {
+          name: { $concat: ['$firstName', ' ', '$lastName'] },
+        },
+      },
+    ])
+      .collation({ locale: 'uk' })
+      .facet({
+        items: aggregatedItems,
+        calculations: [{ $match: filteredItems }, { $count: 'count' }],
+      })
+      .then(data => data[0]);
+
+    const { items, calculations } = users;
+    let usersCount;
+
+    if (calculations && calculations.length) {
```

Comment:
If you made a destructuration, it becomes not useful.

Reply:
fixed